src/inmanta/protocol/rest/__init__.py

```diff
@@ -189,21 +189,30 @@ async def _run_getters(self, arg: str, value: Optional[object]) -> Optional[obje
             LOGGER.exception("Failed to use getter for arg %s", arg)
             raise e
 
-    async def process(self) -> None:
+    @staticmethod
+    def _ensure_list_if_list_type(arg_type: Optional[Type[object]], arg_value: object) -> object:
         """
-        Process the message
+        Handles processing of arguments for GET requests, especially for list types encoded as URL query parameters.
+        If a GET endpoint has a parameter of type list that is encoded as a URL query parameter and the specific request
+        provides a list with one element, urllib doesn't parse it as a list. Map it here explicitly to a list.
         """
-        args: list[str] = list(self._argspec.args)
+        if typing_inspect.is_optional_type(arg_type):
+            non_none_arg_types = [arg for arg in typing_inspect.get_args(arg_type) if arg is not type(None)]
+            if len(non_none_arg_types) == 1 and typing_inspect.get_origin(non_none_arg_types[0]) is list:
+                arg_type = non_none_arg_types[0]
```

Comment:
Do you know why the second condition is required? This if-block only removes the optional part of the type. The code that follow check whether it's a list type or not, right? So I guess this condition is redundant here.
```suggestion
            if len(non_none_arg_types) == 1:
                arg_type = non_none_arg_types[0]
```

Reply:
indeed you are right